lighthouse-core/test/gather/mock-commands.js

```diff
@@ -33,20 +33,37 @@ function createMockSendCommandFn() {
     return Promise.resolve(returnValue);
   });
 
-  mockFn.mockResponse = (command, response, delay) => {
-    mockResponses.push({command, response, delay});
-    return mockFn;
-  };
-
-  mockFn.mockResponseToSession = (command, sessionId, response, delay) => {
-    mockResponses.push({command, sessionId, response, delay});
-    return mockFn;
-  };
-
-  mockFn.findInvocation = (command, sessionId) => {
-    expect(mockFn).toHaveBeenCalledWith(command, sessionId, expect.anything());
-    return mockFn.mock.calls.find(call => call[0] === command && call[1] === sessionId)[2];
-  };
+  const mockFn = Object.assign(mockFnImpl, {
+    /**
+     * @template {keyof LH.CrdpCommands} C
+     * @param {C} command
+     * @param {RecursivePartial<LH.CrdpCommands[C]['returnType']>=} response
```

Comment:
Test code often doesn't provide a full protocol response, so this makes everything optional (yet well-typed for the properties that are given).